proteus/mprans/SpatialTools.py

```diff
@@ -86,6 +88,60 @@ def _attachAuxiliaryVariable(self, key):
                 self.auxiliaryVariables[key] = True
             if key == 'RelaxZones':
                 self.auxiliaryVariables[key] = self.zones
+            if str(key).startswith('Gauge_'):
+                self.auxiliaryVariables[key] = [gauge]
+        elif str(key).startswith('Gauge_') \
+                and gauge not in self.auxiliaryVariables[key]:
+            self.auxiliaryVariables[key] += [gauge]
```

Comment:
True, the logic doesn't really give any warning for failed keys (technically this only should be called by other methods, so tests should be handling the heavy lifting of checking for correct behavior, but some else statements would still be nice).  I'm not sure how hard it should fail (as the function might be called multiple times with the same values, breaking out of the if statement intentionally).

The indentation is correct for the original intention of the function, but it depends on the direction of where we take the logic.  The function was made to only accept the first `RigidBody` and `RelaxZones`.  This might be because the values in `self.auxiliaryVariables` do get used a lot elsewhere and a lot of code is run multiple times.  It could be to keep from attaching the keys multiple times, and maybe messing some stuff up with the `RelaxZones` (that is, resetting it might break things?).  The gauges accept any number of unique gauges, so they needed the `elif` to handle new additions.